File: python/array_string/haystack.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0

        haystack_len, needle_len = len(haystack), len(needle)

        for i in range(haystack_len - needle_len + 1):
            if haystack[i : i + needle_len] == needle:
                return i

        return -1

File: golang/array_string/haystack.go
package main

func strStr(haystack string, needle string) int {
	if needle == "" {
		return 0
	}

	haystack_len, needle_len := len(haystack), len(needle)

	for i := 0; i <= haystack_len-needle_len; i++ {
		if haystack[i:i+needle_len] == needle {
			return i
		}
	}

	return -1
}
